nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -176,548 +159,500 @@ in {
+                The file to store the PID in.
+              '';
+            };
+
+            log_config = mkOption {
+              type = types.path;
+              default = "${cfg.dataDir}/log_config.yaml";
+              readOnly = true;
+              description = ''
+                The file that holds the logging configuration.
+              '';
+            };
+
+            media_store_path = mkOption {
+              type = types.path;
+              default = if lib.versionAtLeast config.system.stateVersion "22.05"
+                then "${cfg.dataDir}/media_store"
+                else "${cfg.dataDir}/media";
+              description = ''
+                Directory where uploaded images and attachments are stored.
+              '';
+            };
+
+            public_baseurl = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "https://example.com:8448/";
+              description = ''
+                The public-facing base URL for the client API (not including _matrix/...)
+              '';
+            };
+
+            tls_certificate_path = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "/var/lib/acme/example.com/fullchain.pem";
+              description = ''
+                PEM encoded X509 certificate for TLS.
+                You can replace the self-signed certificate that synapse
+                autogenerates on launch with your own SSL certificate + key pair
+                if you like.  Any required intermediary certificates can be
+                appended after the primary certificate in hierarchical order.
+              '';
+            };
+
+            tls_private_key_path = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "/var/lib/acme/example.com/key.pem";
+              description = ''
+                PEM encoded private key for TLS. Specify null if synapse is not
+                speaking TLS directly.
+              '';
+            };
+
+            presence.enabled = mkOption {
+              type = types.bool;
+              default = false;
+              example = true;
+              description = ''
+                Whether to enable presence tracking.
+
+                Presence tracking allows users to see the state (e.g online/offline)
+                of other local and remote users.
+              '';
+            };
+
+            listeners = mkOption {
               type = types.listOf (types.submodule {
                 options = {
-                  names = mkOption {
+                  port = mkOption {
+                    type = types.port;
+                    example = 8448;
+                    description = ''
+                      The port to listen for HTTP(S) requests on.
+                    '';
+                  };
+
+                  bind_addresses = mkOption {
```

Comment:
This was migrated from `bind_address` (`types.str`) to `bind_addresses` (`types.listOf types.str`) as per https://github.com/matrix-org/synapse/blob/b7282fe7d17ec767a30725dcb1f66107ef0bd838/docs/sample_config.yaml#L255-L256.

Now the problem with that is that while an empty list would cause synapse to listen on "all interfaces", but that breaks the assumptions made in `registerNewMatrixUser`.

I would appreciate suggestions on how to tackel this.

Reply:
So if the list is empty it doesn't matter which address we use? So how about a fallback like `(head (bind_addresses ++ ["localhost"]))` then?

Reply:
On a second glance, `registerNewMatrixUser` has some more serious issues with making strong assumptions about the config. Maybe we should remove it again?

Reply:
I would be fine with removing it again.

Reply:
I already made some changes, PTAL.

[`e9c74b5` (#158605)](https://github.com/NixOS/nixpkgs/pull/158605/commits/e9c74b57e5e55b34aa55f562bc86d9ebfe2f59f5)

Reply:
Great, thanks!